e2e/etcd_test.go

```diff
@@ -103,6 +109,108 @@ func testProcessClusterPutGet(t *testing.T, cfg *etcdProcessClusterConfig) {
 	}
 }
 
+func TestBasicOpsV2CtlWatchWithProxy(t *testing.T) {
+	testProcessClusterV2CtlWatch(
+		t,
+		&etcdProcessClusterConfig{
+			clusterSize:  3,
+			proxySize:    1,
+			isClientTLS:  false,
+			isPeerTLS:    false,
+			initialToken: "new",
+		},
+		etcdProcessBasePort+40,
+		false,
+		"foo", "bar",
+	)
+}
+
+func TestBasicOpsV2CtlWatchWithProxyNoSync(t *testing.T) {
+	testProcessClusterV2CtlWatch(
+		t,
+		&etcdProcessClusterConfig{
+			clusterSize:  3,
+			proxySize:    1,
+			isClientTLS:  false,
+			isPeerTLS:    false,
+			initialToken: "new",
+		},
+		etcdProcessBasePort+50,
+		true,
+		"foo", "bar",
+	)
+}
+
+func testProcessClusterV2CtlWatch(t *testing.T, cfg *etcdProcessClusterConfig, basePort int, noSync bool, key, value string) {
+	if fileutil.Exist("../bin/etcdctl") == false {
+		t.Fatalf("could not find etcdctl binary")
+	}
+
+	epc, errC := newEtcdProcessCluster(cfg, basePort)
+	if errC != nil {
+		t.Fatalf("could not start etcd process cluster (%v)", errC)
+	}
+	defer func() {
+		if errC := epc.Close(); errC != nil {
+			t.Fatalf("error closing etcd processes (%v)", errC)
+		}
+	}()
+
+	endpoint, be := "", ""
+	for _, p := range epc.procs {
+		if p.cfg.isProxy {
+			endpoint = p.cfg.acurl.String()
+			time.Sleep(5 * time.Second) // give some time for proxy to find cluster
```

Comment:
this is racey; instead, `newEtcdProcessCluster` shouldn't return until the proxies are ready